dds/DCPS/DataReaderImpl_T.h

```diff
@@ -915,6 +915,22 @@ namespace OpenDDS {
       if (!encap.to_encoding(encoding, MarshalTraitsType::extensibility())) {
         return;
       }
+
+      if (decoding_modes_.find(encoding.kind()) == decoding_modes_.end()) {
+        ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR ")
+          ACE_TEXT("%CDataReaderImpl::lookup_instance: ")
+          ACE_TEXT("Encoding kind of the received sample (%C) does not ")
+          ACE_TEXT("match the ones specified by DataReader.\n"),
+          TraitsType::type_name(),
+          Encoding::kind_to_string(encoding.kind()).c_str()));
```

Comment:
Looking at this, it would probably be better as a warning that depends on `DCPS_debug_level`.